packages/@aws-cdk/aws-backup/lib/selection.ts

```diff
@@ -85,12 +118,40 @@ export class BackupSelection extends Resource implements iam.IGrantable {
     const role = props.role || new iam.Role(this, 'Role', {
       assumedBy: new iam.ServicePrincipal('backup.amazonaws.com'),
     });
-    role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSBackupServiceRolePolicyForBackup'));
-    if (props.allowRestores) {
-      role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSBackupServiceRolePolicyForRestores'));
+
+    // Only add managed policies if the customRole field is false
+    if (!props.customRole) {
+      role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSBackupServiceRolePolicyForBackup'));
+      if (props.allowRestores) {
+        role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSBackupServiceRolePolicyForRestores'));
+      }
+
+      // Conditionally add S3 specific managed policies (Backup and Restores)
+      if (props.allowS3Backup) {
+        role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AWSBackupServiceRolePolicyForS3Backup'));
+      }
+      if (props.allowS3Restores) {
+        role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AWSBackupServiceRolePolicyForS3Restore'));
+      }
+    } else {
+      // Provide deployment warnings if customRole is enabled but any of the
+      // permission flags are true
+      if (props.allowRestores) {
+        Annotations.of(this).addWarning('Both the customRole and allowRestores flag was set to true - no restore managed policy was added.');
+      }
+      if (props.allowS3Backup) {
+        Annotations.of(this).addWarning('Both the customRole and allowS3Backup flag was set to true - no S3 backup managed policy was added.');
+      }
+      if (props.allowS3Restores) {
+        Annotations.of(this).addWarning('Both the customRole and allowS3Restores flag was set to true - no S3 restore managed policy was added.');
+      }
```

Comment:
Should the customer role not be set somewhere though, as per
>So I added a customRole flag which can be provided to disable the addition of any managed policies, meaning that a user could provide a completely custom role, if desired.

Reply:
Not sure what you mean here - I think maybe `customRole` is not a good name for the field. 

Personally I think the default behaviour of the service role policy being added even when the user provides their own Role is debatable/opinionated - this flag is to add the capacity to have the construct just reference your provided role or create one on your behalf without adding any policies to it. 

Happy for ideas on how to clarify this - but as I said I don't fully understand your comment above.

Reply:
Maybe another approach would be to add an `allowBackup` flag (similar to `allowRestore`) which is default to `true`. Then all of the managed policies can just be switched on/off, but a reasonable default selection of policies is present. This also means there would be no breaking change to the API.